source/common/http/http2/codec_impl.cc

```diff
@@ -25,6 +25,32 @@ namespace Envoy {
 namespace Http {
 namespace Http2 {
 
+class Http2ResponseCodeDetailValues {
+  // Invalid HTTP header field was received and stream is going to be
+  // closed.
+  const absl::string_view NgHttp2ErrHttpHeader = "http2.invalid.header.field";
+
+  // Violation in HTTP messaging rule.
+  const absl::string_view NgHttp2ErrHttpMessaging = "http2.violation.of.messaging.rule";
+
+  // none of the above
+  const absl::string_view NgHttp2ErrUnknown = "http2.unknown.nghttp2.error";
+
+public:
+  const absl::string_view strerror(int error_code) const {
+    switch (error_code) {
+    case NGHTTP2_ERR_HTTP_HEADER:
+      return NgHttp2ErrHttpHeader;
+    case NGHTTP2_ERR_HTTP_MESSAGING:
+      return NgHttp2ErrHttpMessaging;
+    default:
+      return NgHttp2ErrUnknown;
```

Comment:
Do you have coverage for this case?

Reply:
The two explicit cases are hit in various places in the unit and integration tests.

I don't have a test for the default case, and I am not sure whether it can be triggered.  The theory we discussed with Alyssa was to return unknown, and whoever sees unknown in future logs will dig into it and add another explicit case.

Reply:
I think that's fine, but generally there should be some kind of coverage around this, perhaps by making the translation functions static and making sure they do the right thing during targeted unit testing.

Reply:
@mattklein123 I can add the test that are you asking for, but I see little value in it, since it will look like a duplication of the switch statement, something like:

check(NGTTP2_ERR_..., "foo");
check(NGTTP2_ERR_..., "bar");
check(-1, "quux");

My inclination is to close this as WONTFIX.  If you have other concerns (e.g., you are striving for 100% coverage and this code gets in the way), please let me know and I'll be happy to oblige.

Reply:
It doesn't thrill me that we have code that we are not sure whether we can ever hit, and we have no test coverage for it either. I agree though that this type of test is not super useful. Up to you.